feat(core/cli): add "list" command for the CLI to list all installed formulas and casks.

tommyming opened the pull request:
This is a draft PR that aims to solve the request of #43. 

### Why create this PR?
Currently, there is no command that allows user to check installed formulas and casks. Therefore I added a command handle this kind of situation. I am also planning to add indicators 

### Why it is a draft?
There are several reasons to make it as a draft, instead of a proper PR:
- the core part of codebase is currently under active development and refactoring (which is definitely having a higher priority in my opinion.)
- I am still not sure if "list" is a good name for this command (Yeah, the naming issues.)
- I want to add and refine a bit more on the current code.

### Usage
- `sps list`: It will show all the formulas and casks, with their latest installed version.
- `sps list --all`: With the `all` flag, all the installed formulas and casks will be shown.

Feel free to give comments and review the code!

alexykn commented:
I'll take a look at this tonight after work. Thank you!

tommyming commented:
I still got refactoring work going, maybe will update this weekend.

alexykn commented:
All right, I'll not yet merge this then. Glancing over the code it looks good. Just tell me when it's ready.

tommyming commented:
I have removed the "all" command, and add "formula" and "cask" command instead, I think the "all" command is not that useful.

Have also done some refactoring and updated README.

Will change to "ready" state once I capture some pic and upload here!

tommyming commented:
Expected result of commands:
- `sps list`
<img width="561" alt="Screenshot 2025-05-10 at 20 40 21" src="https://github.com/user-attachments/assets/0dfaace3-7612-4c62-8824-b639eaf18621" />

- `sps list --formula`
<img width="561" alt="Screenshot 2025-05-10 at 20 40 38" src="https://github.com/user-attachments/assets/f5a3f470-f020-40af-ac3c-69639d2f1775" />

- `sps list --cask` (Nothing case, if you have no casks installed)
<img width="731" alt="Screenshot 2025-05-10 at 20 40 55" src="https://github.com/user-attachments/assets/05b85678-8b2d-45df-8176-e532cb9ab6a6" />

#### Potential Problem
I currently can't get the installed casks by using the `get_installed_packages()` method, might still need to check.

alexykn commented:
looks good, thank you!
